### ISSUE TITLE
We are facing a issue with Optimized Markup Features

### ISSUE THREAD

**Issue by bdkoder** [elementor/elementor#29610]
### Prerequisites

- [X] I have searched for similar issues in open and closed tickets and cannot find a duplicate.
- [X] I have troubleshooted my issue, and it still exists against the latest stable version of Elementor.

### Description

If we disabled the feature / set to default then our Extension features work well.
But when we Active the feature the data-settings attribute missing from the DOM and the extensions are not working.

Thanks for your understanding & kind patience. 

![image](https://github.com/user-attachments/assets/b5c67253-6dac-45a5-8550-896c96801e3e)

Here is a video  - https://www.loom.com/share/7aa874fab2914faf968ee915ea92f88d?sid=63110e33-938b-40b9-b526-79f549b2b671

Wish, will help you.

### Steps to reproduce

Here is a video  - https://www.loom.com/share/7aa874fab2914faf968ee915ea92f88d?sid=63110e33-938b-40b9-b526-79f549b2b671

### Expected behavior

Extensions / Features should work properly.

### Isolating the problem

- [ ] This bug happens when only the Elementor (and Elementor Pro) plugins are active.
- [ ] This bug happens with the Hello Elementor theme active.
- [ ] I can reproduce this bug consistently by following the steps I described above.

### Elementor System Info

```txt
== Server Environment ==
	Operating System: Darwin
	Software: nginx/1.16.0
	MySQL version: MySQL Community Server - GPL v8.0.16
	PHP Version: 8.3.11
	PHP Memory Limit: 1000M
	PHP Max Input Vars: 4000
	PHP Max Post Size: 1000M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Not connected (cURL error 28: Operation timed out after 5002 milliseconds with 10420 bytes received)

== WordPress Environment ==
	Version: 6.7.1
	Site URL: http://192.168.1.111
	Home URL: http://192.168.1.111
	WP Multisite: No
	Max Upload Size: 300 MB
	Memory limit: 1000M
	Max Memory limit: 1000M
	Permalink Structure: /%postname%/
	Language: en_US
	Timezone: Asia/Dhaka
	Admin Email: dev-email@wpengine.local
	Debug Mode: Active

== Theme ==
	Name: Astra
	Version: 4.8.8
	Author: Brainstorm Force
	Child Theme: No

== User ==
	Role: administrator
	WP Profile lang: en-US
	User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/131.0.0.0 Safari/537.36

== Active Plugins ==
	Data Collector Insights
		Version: 1.5.0
		Author: Shahidul Islam

	Elementor
		Version: 3.26.0
		Author: Elementor.com

	Element Pack Pro
		Version: 7.18.13
		Author: BdThemes

	HTTP Requests Manager
		Version: 1.3.6
		Author: veppa

	Transients Manager
		Version: 2.0.7
		Author: WPBeginner

	WooCommerce
		Version: 9.4.3
		Author: Automattic

	WP Crontrol
		Version: 1.17.1
		Author: John Blackbourn


== Elements Usage ==
	
	container : 2
		bdt-advanced-button : 1
		bdt-post-gallery : 1
		bdt-scrollnav : 1
		container : 5
		image : 1
	page : 5
		bdt-advanced-button : 11
		bdt-advanced-gmap : 1
		bdt-advanced-heading : 36
		bdt-advanced-icon-box : 10
		bdt-brand-grid : 1
		bdt-custom-carousel : 2
		bdt-fancy-list : 2
		bdt-flip-box : 4
		bdt-image-accordion : 1
		bdt-image-expand : 1
		bdt-mailchimp : 1
		bdt-marquee : 8
		bdt-navbar : 1
		bdt-panel-slider : 1
		bdt-post-block : 1
		bdt-product-carousel : 2
		bdt-review-card-grid : 2
		bdt-scrollnav : 4
		bdt-search : 2
		bdt-sub-menu : 9
		bdt-tabs : 18
		container : 107
		divider : 2
		heading : 21
		icon-list : 6
		image : 44
		lightbox : 1
		social-icons : 3
		text-editor : 35
	wp-post : 0
		bdt-navbar : 1
		container : 0
		heading : 1
	wp-page : 12
		bdt-advanced-button : 5
		bdt-advanced-calculator : 1
		bdt-advanced-heading : 13
		bdt-advanced-icon-box : 7
		bdt-animated-heading : 1
		bdt-carousel : 1
		bdt-contact-form : 1
		bdt-countdown : 2
		bdt-custom-carousel : 1
		bdt-custom-gallery : 1
		bdt-horizontal-scroller : 1
		bdt-hover-video : 3
		bdt-lottie-icon-box : 1
		bdt-mailchimp : 1
		bdt-marquee : 7
		bdt-member : 7
		bdt-modal : 2
		bdt-post-gallery : 1
		bdt-price-list : 2
		bdt-progress-pie : 4
		bdt-review-card-carousel : 1
		bdt-scroll-button : 1
		bdt-scrollnav : 2
		bdt-slideshow : 1
		bdt-sub-menu : 4
		bdt-webhook-form : 1
		button : 7
		container : 85
		heading : 25
		icon-box : 4
		image : 18
		prime-slider-blog : 1
		social-icons : 2
		spacer : 1
		text-editor : 17


== Settings ==
	
	cpt_support: post, page, product
	allow_tracking: yes
	unfiltered_files_upload: 1
	font_display: swap


== Elementor Experiments ==
	Inline Font Icons: Active by default
	Additional Custom Breakpoints: Active by default
	Container: Active
	Upgrade Swiper Library: Active by default
	Nested Elements Performance: Active by default
	Optimized Markup: Active
	Plugin Onboarding: Active by default
	CSS Smooth Scroll: Active by default
	Elementor Home Screen: Active by default
	Landing Pages: Inactive by default
	Nested Elements: Inactive
	Editor Top Bar: Active by default
	Pages Panel: Inactive by default
	Element Caching: Inactive
	Link In Bio: Active by default
	Floating Buttons: Active by default
	Elementor Editor Events: Inactive by default
	Atomic Widgets: Inactive by default
	Global Classes: Inactive by default
	Launchpad Checklist: Inactive by default


== Log ==
	
Log: showing 20 of 372024-11-26 10:22:14 [info] Elementor/Upgrades - _on_each_version Start  
2024-11-26 10:22:14 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.25.10',
  'to' => '3.26.0-beta1',
)]
2024-11-26 10:22:16 [info] Elementor/Upgrades - _on_each_version Finished 
2024-11-26 10:22:16 [info] Elementor/Upgrades - _v_3_26_0 Start  
2024-11-26 10:22:16 [info] Elementor/Upgrades - _v_3_26_0 Finished 
2024-11-26 10:22:16 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.25.10',
  'to' => '3.26.0-beta1',
)]
2024-12-14 15:19:46 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.26.0-beta1',
  'to' => '3.26.0-beta5',
)]
2024-12-14 15:19:46 [info] elementor::elementor_updater Started 
2024-12-14 15:19:46 [info] Elementor/Upgrades - _on_each_version Start  
2024-12-14 15:19:48 [info] Elementor/Upgrades - _on_each_version Finished 
2024-12-14 15:19:48 [info] Elementor/Upgrades - _v_3_26_0 Start  
2024-12-14 15:19:48 [info] Elementor/Upgrades - _v_3_26_0 Finished 
2024-12-14 15:19:48 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.26.0-beta1',
  'to' => '3.26.0-beta5',
)]
2024-12-17 10:40:36 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.26.0-beta5',
  'to' => '3.26.0',
)]
2024-12-17 10:40:36 [info] elementor::elementor_updater Started 
2024-12-17 10:40:36 [info] Elementor/Upgrades - _on_each_version Start  
2024-12-17 10:40:43 [info] Elementor/Upgrades - _on_each_version Finished 
2024-12-17 10:40:43 [info] Elementor/Upgrades - _v_3_26_0 Start  
2024-12-17 10:40:43 [info] Elementor/Upgrades - _v_3_26_0 Finished 
2024-12-17 10:40:43 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.26.0-beta5',
  'to' => '3.26.0',
)]

JS: showing 4 of 4JS: 2024-11-23 09:15:48 [error X 1][http://192.168.1.111/wp-content/plugins/elementor/assets/js/frontend-modules.js?ver=3.25.9:641:31] Cannot read properties of undefined (reading 'attributes') 
JS: 2024-11-26 05:52:20 [error X 4][http://192.168.1.111/wp-content/plugins/elementor/assets/js/editor.js?ver=3.26.0-beta1:40465:71] Cannot read properties of undefined (reading 'toLowerCase') 
JS: 2024-12-07 06:13:53 [error X 1][http://192.168.1.111/wp-content/plugins/elementor/assets/js/editor.js?ver=3.26.0-beta1:31948:31] Cannot read properties of undefined (reading 'isDesignable') 
JS: 2024-12-11 04:43:22 [error X 2][http://192.168.1.111/wp-content/plugins/bdthemes-prime-slider-lite/assets/js/modules/ps-general.min.js?ver=3.15.23:1:107] Cannot read properties of undefined (reading 'id') 

PHP: showing 2 of 2PHP: 2024-12-03 15:20:21 [error X 25][/Users/shahidul/Local Sites/ep-dev/app/public/wp-content/plugins/elementor/core/files/css/base.php::449] Uncaught TypeError: array_intersect_key(): Argument #1 ($array) must be of type array, true given in /Users/shahidul/Local Sites/ep-dev/app/public/wp-content/plugins/elementor/core/files/css/base.php:449
Stack trace:
#0 /Users/shahidul/Local Sites/ep-dev/app/public/wp-content/plugins/elementor/core/files/css/base.php(449): array_intersect_key(true, Array)
#1 /Users/shahidul/Local Sites/ep-dev/app/public/wp-content/plugins/elementor/core/files/css/base.php(726): Elementor\Core\Files\CSS\Base->add_control_rules(Array, Array, Object(Closure), Array, Array, Array)
#2 /Users/shahidul/Local Sites/ep-dev/app/public/wp-content/plugins/elementor/core/files/css/base.php(592): Elementor\Core\Files\CSS\Base->add_control_style_rules(Array, Array, Array, Array, Array)
#3 /Users/shahidul/Local Sites/ep-dev/app/public/wp-content/plugins/elementor/core/files/css/post.php(222): Elementor\Core\Files\CSS\Base->add_controls_stack_style_rules(Object(ElementPack\Modules\Member\Widgets\Member), Array, Array, Array, Array, Array)
#4 /Users/shahidul/Local Sites/ep-dev/app/public/wp-content/plugins/elementor/core/files/css/post.php(313): Elementor\Core\Files\CSS\Post->add_controls_stack_style_rules(Object(ElementPack\Modules\Member\Widgets\Member), Array, Array, Array, Array)
#5 /Users/shahidul/Local Sites/ep-dev/app/public/wp-content/plugins/elementor/core/files/css/post.php(297): Elementor\Core\Files\CSS\Post->render_element_styles(Object(ElementPack\Modules\Member\Widgets\Member))
#6 /Users/shahidul/Local Sites/ep-dev/app/public/wp-content/plugins/elementor/core/files/css/post.php(226): Elementor\Core\Files\CSS\Post->render_styles(Object(ElementPack\Modules\Member\Widgets\Member))
#7 /Users/shahidul/Local Sites/ep-dev/app/public/wp-content/plugins/elementor/core/files/css/post.php(313): Elementor\Core\Files\CSS\Post->add_controls_stack_style_rules(Object(Elementor\Includes\Elements\Container), Array, Array, Array, Array)
#8 /Users/shahidul/Local Sites/ep-dev/app/public/wp-content/plugins/elementor/core/files/css/post.php(297): Elementor\Core\Files\CSS\Post->render_element_styles(Object(Elementor\Includes\Elements\Container))
#9 /Users/shahidul/Local Sites/ep-dev/app/public/wp-content/plugins/elementor/core/files/css/post.php(226): Elementor\Core\Files\CSS\Post->render_styles(Object(Elementor\Includes\Elements\Container))
#10 /Users/shahidul/Local Sites/ep-dev/app/public/wp-content/plugins/elementor/core/files/css/post.php(313): Elementor\Core\Files\CSS\Post->add_controls_stack_style_rules(Object(Elementor\Includes\Elements\Container), Array, Array, Array, Array)
#11 /Users/shahidul/Local Sites/ep-dev/app/public/wp-content/plugins/elementor/core/files/css/post.php(297): Elementor\Core\Files\CSS\Post->render_element_styles(Object(Elementor\Includes\Elements\Container))
#12 /Users/shahidul/Local Sites/ep-dev/app/public/wp-content/plugins/elementor/core/files/css/post.php(178): Elementor\Core\Files\CSS\Post->render_styles(Object(Elementor\Includes\Elements\Container))
#13 /Users/shahidul/Local Sites/ep-dev/app/public/wp-content/plugins/elementor/core/files/css/base.php(687): Elementor\Core\Files\CSS\Post->render_css()
#14 /Users/shahidul/Local Sites/ep-dev/app/public/wp-content/plugins/elementor/core/files/base.php(180): Elementor\Core\Files\CSS\Base->parse_content()
#15 /Users/shahidul/Local Sites/ep-dev/app/public/wp-content/plugins/elementor/core/files/css/base.php(133): Elementor\Core\Files\Base->update_file()
#16 /Users/shahidul/Local Sites/ep-dev/app/public/wp-content/plugins/elementor/core/files/css/base.php(229): Elementor\Core\Files\CSS\Base->update()
#17 /Users/shahidul/Local Sites/ep-dev/app/public/wp-content/plugins/elementor/core/files/css/post.php(201): Elementor\Core\Files\CSS\Base->enqueue()
#18 /Users/shahidul/Local Sites/ep-dev/app/public/wp-content/plugins/elementor/includes/frontend.php(695): Elementor\Core\Files\CSS\Post->enqueue()
#19 /Users/shahidul/Local Sites/ep-dev/app/public/wp-includes/class-wp-hook.php(324): Elementor\Frontend->enqueue_styles('')
#20 /Users/shahidul/Local Sites/ep-dev/app/public/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters(NULL, Array)
#21 /Users/shahidul/Local Sites/ep-dev/app/public/wp-includes/plugin.php(517): WP_Hook->do_action(Array)
#22 /Users/shahidul/Local Sites/ep-dev/app/public/wp-includes/script-loader.php(2272): do_action('wp_enqueue_scri...')
#23 /Users/shahidul/Local Sites/ep-dev/app/public/wp-includes/class-wp-hook.php(324): wp_enqueue_scripts('')
#24 /Users/shahidul/Local Sites/ep-dev/app/public/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters(NULL, Array)
#25 /Users/shahidul/Local Sites/ep-dev/app/public/wp-includes/plugin.php(517): WP_Hook->do_action(Array)
#26 /Users/shahidul/Local Sites/ep-dev/app/public/wp-includes/general-template.php(3064): do_action('wp_head')
#27 /Users/shahidul/Local Sites/ep-dev/app/public/wp-content/themes/astra/header.php(31): wp_head()
#28 /Users/shahidul/Local Sites/ep-dev/app/public/wp-includes/template.php(810): require_once('/Users/shahidul...')
#29 /Users/shahidul/Local Sites/ep-dev/app/public/wp-includes/template.php(745): load_template('/Users/shahidul...', true, Array)
#30 /Users/shahidul/Local Sites/ep-dev/app/public/wp-includes/general-template.php(48): locate_template(Array, true, true, Array)
#31 /Users/shahidul/Local Sites/ep-dev/app/public/wp-content/themes/astra/page.php(20): get_header()
#32 /Users/shahidul/Local Sites/ep-dev/app/public/wp-includes/template-loader.php(106): include('/Users/shahidul...')
#33 /Users/shahidul/Local Sites/ep-dev/app/public/wp-blog-header.php(19): require_once('/Users/shahidul...')
#34 /Users/shahidul/Local Sites/ep-dev/app/public/index.php(17): require('/Users/shahidul...')
#35 {main}
  thrown [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2024-12-05 15:56:31 [error X 1][/Users/shahidul/Local Sites/ep-dev/app/public/wp-content/plugins/elementor/modules/ai/module.php::231] Uncaught Error: Call to a member function get_gallery_image_ids() on false in /Users/shahidul/Local Sites/ep-dev/app/public/wp-content/plugins/elementor/modules/ai/module.php:231
Stack trace:
#0 /Users/shahidul/Local Sites/ep-dev/app/public/wp-includes/class-wp-hook.php(324): Elementor\Modules\Ai\Module->get_product_images_ajax('')
#1 /Users/shahidul/Local Sites/ep-dev/app/public/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters('', Array)
#2 /Users/shahidul/Local Sites/ep-dev/app/public/wp-includes/plugin.php(517): WP_Hook->do_action(Array)
#3 /Users/shahidul/Local Sites/ep-dev/app/public/wp-admin/admin-ajax.php(192): do_action('wp_ajax_element...')
#4 {main}
  thrown [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]



== Elementor - Compatibility Tag ==
	
	Element Pack Pro: Compatible
```

### Agreement

- [X] I agree that my issue may be closed without action if it doesn't meet all the requirements.

**Comment by maudud-bdthemes** [comment]
I am facing this same issue. It should be resolved as soon as possible. Please look into this issue.

**Comment by bddevfarid** [comment]
Also facing the same issue. Please resolve it as soon as possible.

**Comment by louiswol94** [comment]
Hi everyone, we merged a fix regarding this issue. The `data-settings` attribute should return now. You still need to check your JavaScript if you applied the Tooltip on the `elementor-widget-container` specifically it might still not work even if the data-settings is there. Your code should work whether the `elementor-widget-container` div is present or not. Thanks!

**Comment by nicholaszein** [comment]
This issue was resolved with the release of Elementor Core v3.26.2.
